var/spack/repos/builtin/packages/farmhash/package.py

```diff
@@ -0,0 +1,42 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os.path
+from shutil import copyfile
+
+
+class Farmhash(CMakePackage):
+    """
+    FarmHash provides hash functions for strings and other data.
+    """
+
+    homepage = "https://github.com/google/farmhash"
+
+    version('92e897', git='https://github.com/google/farmhash.git', 
+            commit='92e897b282426729f4724d91a637596c7e2fe28f')
+
+    def patch(self):
+        copyfile(join_path(os.path.dirname(__file__), "CMakeLists.txt"),
```

Comment:
I think we can actually convert this one to an autotools build.  The CMakeLists.txt is from the contributed cmake build in Tensorflow (and lives in its own kind of weird directory there), but it seems like it's not always maintained well.

I'd rather convert this to use the package's native build system (autotools) than keep relying on the external CMakeLists.txt file.

I have he same question as @davydden for the TF cmake maintainers though -- seems like they could submit this upstream, or just use the native build with ExternalPackage_Add.  I am not sure why they don't.